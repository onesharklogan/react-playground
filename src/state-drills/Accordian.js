import React from 'react';
import ReactDOM from 'react-dom';
import './Accordian.css';

class Accordian extends React.Component {
    static defaultProps = { sections: [] };

    state = {
        activeSectionIndex: null
    }

    handleButtonClick = (indexChosen) => {
        console.log("Pressed button");
        this.setState({
            activeSectionIndex: indexChosen
        })
    }

    renderAccordionItems() {
        return this.props.sections.map((section, i) => (

            <li key={i}>
                <button key={i} onClick={() => this.handleButtonClick(i)}>{this.props.sections[i].title}</button>
                <p>{i == this.state.activeSectionIndex && this.props.sections[i].content}
                </p>
            </li>
        ))
    }



    render(props) {
        return (
            <div>
                <ul >
                    {this.renderAccordionItems()}
                </ul>
            </div>
        )
    }
}

export default Accordian;