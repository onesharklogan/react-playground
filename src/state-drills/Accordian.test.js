import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';
import { Simulate } from 'react-dom/test-utils';


const sections = [
    {
        title: 'Section 1',
        content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
        title: 'Section 2',
        content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
        title: 'Section 3',
        content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
]

it('renders empty given no Sections', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders no sections by default', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('opens any clicked section', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})

it('opens only one section at a time', () => {
    const wrapper = shallow(<Accordian sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')

    expect(toJson(wrapper)).toMatchSnapshot()
})