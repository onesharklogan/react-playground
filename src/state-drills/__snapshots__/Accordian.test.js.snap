// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens any clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`opens only one section at a time 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;

exports[`renders empty given no Sections 1`] = `
<div>
  <ul />
</div>
`;

exports[`renders no sections by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
      <p />
    </li>
    <li
      key="1"
    >
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
      <p />
    </li>
    <li
      key="2"
    >
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
      <p />
    </li>
  </ul>
</div>
`;
